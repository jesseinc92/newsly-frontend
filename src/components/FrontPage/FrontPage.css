.FrontPage h1 {
  font-family: 'Fjalla One', sans-serif;
}

.page-count p {
  display: inline-block;
}