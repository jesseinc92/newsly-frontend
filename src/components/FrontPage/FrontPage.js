import React, { useEffect, useState } from 'react';
import usePagination from '../../hooks/usePagination';
import NewslyAPI from '../../api';
import ArticleCard from '../ArticleCard/ArticleCard';
import Button from '../Button/Button';
import './FrontPage.css';

const FrontPage = () => {
  const [articles, setArticles] = useState([]);
  const [pageNum, handlePagination] = usePagination();

  useEffect(() => {
    const getArticles = async () => {
      const newestArticles = await NewslyAPI.getNewestArticles(pageNum);
      setArticles(newestArticles);
    }
    getArticles();

    // scroll to the top of the page on every change
    window.scrollTo({ 
      top: 0, 
      left: 0, 
      behavior: 'smooth' 
    });
  }, [pageNum]);

  return (
    <main className='FrontPage'>
      <h1>Front Page</h1>

      <section className='FrontPage-articles'>
        {articles.length > 0 ?
          articles.map(article => (
            <ArticleCard key={article.id} article={article} />
          ))
          :
          <p>Loading...</p>
        } 
      </section>

      <div className='button-group page-count'>
          <Button text='previous' handler={handlePagination} />
          <p>Page { pageNum }</p>
          <Button text='next' handler={handlePagination} />
      </div>
    </main>
  )
}

export default FrontPage;