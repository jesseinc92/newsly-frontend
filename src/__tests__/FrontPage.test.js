import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { render, screen } from '@testing-library/react';
import FrontPage from '../components/FrontPage/FrontPage';

it('renders the component', function() {
  render(<MemoryRouter><FrontPage /></MemoryRouter>)
});

it('matches the snapshot', function() {
  const { asFragment } = render(<MemoryRouter><FrontPage /></MemoryRouter>);
  expect(asFragment()).toMatchSnapshot();
})

it('contains the front page header', function() {
  render(<MemoryRouter><FrontPage /></MemoryRouter>);
  const header = screen.getByText('Front Page');
  expect(header).toBeInTheDocument();
})