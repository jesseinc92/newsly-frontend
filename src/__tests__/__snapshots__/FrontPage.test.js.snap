// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
<DocumentFragment>
  <main
    class="FrontPage"
  >
    <h2>
      Front Page
    </h2>
    <section
      class="FrontPage-articles-wrapper"
    >
      <p
        class="loader-placeholder"
      >
        Loading...
      </p>
    </section>
    <div
      class="button-group page-count"
    >
      <button
        class="Button"
      >
        <a
          href="/"
        >
          previous
        </a>
      </button>
      <p>
        Page 1
      </p>
      <button
        class="Button"
      >
        <a
          href="/"
        >
          next
        </a>
      </button>
    </div>
  </main>
</DocumentFragment>
`;
